tests/core_ptl/check_for_ranks.py

```diff
@@ -83,7 +83,9 @@ def instantiate_multinode_ddp_if_possible():
     num_gpus = torch.cuda.device_count()
     ## Change logger=None to logger=False to support PTL 2.0
     trainer = Trainer(devices=num_gpus, accelerator='gpu', strategy='ddp', logger=False, enable_checkpointing=False)
-    exp_manager_cfg = ExpManagerConfig(exp_dir='./ddp_check/', use_datetime_version=False, version="")
+    exp_manager_cfg = ExpManagerConfig(
+        exp_dir='./ddp_check/', use_datetime_version=False, version="", explicit_log_dir="./ddp_check/log_dir"
```

Comment:
I don't think there is a race condition here, or else it would have popped up at least a few times in the last 3 years of the test. Why is explicit logdir being used ?

Reply:
rank 0 has a logger https://github.com/NVIDIA/NeMo/blob/7de8f87e7637a51461159c3f00295e72c3609904/nemo/utils/exp_manager.py#L782 which makes log_dir `./ddp_check/default/version_0` but rank 1 doesn't have a logdir, which makes log_dir `./ddp_check/default` without the version 0 then this test fails.

if we set explicit log dir here: https://github.com/NVIDIA/NeMo/blob/7de8f87e7637a51461159c3f00295e72c3609904/nemo/utils/exp_manager.py#L740 it short circuits this logic

Reply:
We avoid explicit log dir because it circumvents normal operation of exp  Manager and leads to bugs going undetected. 

You mean to say, over 2.5 years of testing and without fail the race condition always executed and therefore this code ran without fail for that long ? Doesn't seem likely. In either case, id suggest removing explicit log dir and updating the test to work without explicit log dir.